### PR TITLE
Add a target indicator to turrets

### PR DESCRIPTION
**Pull request by ThePythonGuy3**
Basically [this.](https://discord.com/channels/391020510269669376/663853133587546138/761961546061709333)

### CODE REVIEW

**Comment by SnakkiZXZ** [comment]
I think this is a good idea, but it must be dependent on the graphics settings so that it can be disabled

**Comment by ThePythonGuy3** [comment]
why should it be disabled? Its not like it would drain your fps

**Comment by SnakkiZXZ** [comment]
the problem is not in the FPS indicators, but in the fact that it may be inappropriate for some people or interfere with large waves of enemies

**Comment by ThePythonGuy3** [comment]
*theres only one target at a time, bruh*

**Review by SnakkiZXZ — APPROVED** [review]

**Comment by SnakkiZXZ** [comment]
Oh, sorry there was a small mistake when I wanted to suggest a change to you, I hope this will not affect your request in any way

**Comment by ThePythonGuy3** [comment]
its fine. nothing happened, dont worry

**Comment by SnakkiZXZ** [comment]
ok, but you can still make the changes yourself, I suggest this:
```if(target != null){```
replaced by:
```if(target != null && Core.settings.getBool("blockstatus")){```
since the pointer to the target is the status of the turret, we will assign them to the status of the block.

I'm just suggesting whether you decide to do or not.

File: core/src/mindustry/world/blocks/defense/turrets/Turret.java [hunk 1]
```diff
@@ -277,6 +278,14 @@ public void updateTile(){
         @Override
         public void drawSelect(){
             Drawf.dashCircle(x, y, range, team.color);
+            if(target != null) {
```
Comment (summetdev) [inline core/src/mindustry/world/blocks/defense/turrets/Turret.java:281]:
```suggestion
            if(target != null){
```

**Comment by ThePythonGuy3** [comment]
> ok, but you can still make the changes yourself, I suggest this:
> `if(target != null){`
> replaced by:
> `if(target != null && Core.settings.getBool("blockstatus")){`
> since the pointer to the target is the status of the turret, we will assign them to the status of the block.
> 
> I'm just suggesting whether you decide to do or not.

hmm good idea

File: core/src/mindustry/world/blocks/defense/turrets/Turret.java [hunk 2]
```diff
@@ -277,6 +278,14 @@ public void updateTile(){
...
         public void drawSelect(){
             Drawf.dashCircle(x, y, range, team.color);
+            if(target != null && Core.settings.getBool("blockstatus")){
+                float time = Time.time();
+                Lines.stroke(3f, Pal.gray);
+                Lines.square(target.getX(), target.getY(), 8 + 1f + Mathf.sinDeg(time*2), 45+time);
+                Lines.stroke(1f, Team.crux.color);
+                Lines.square(target.getX(), target.getY(), 8 + 1f + Mathf.sinDeg(time*2), 45+time);
+                Draw.reset();
+            }
```
Comment (MEEPofFaith) [inline core/src/mindustry/world/blocks/defense/turrets/Turret.java:288]: The `float time = Time.time();` isn't nessesarry.
```suggestion
            if(target != null && Core.settings.getBool("blockstatus")){
                Lines.stroke(3f, Pal.gray);
                Lines.square(target.getX(), target.getY(), 8 + 1f + Mathf.sinDeg(Time.time() * 2), 45 + Time.time());
                Lines.stroke(1f, Team.crux.color);
                Lines.square(target.getX(), target.getY(), 8 + 1f + Mathf.sinDeg(Time.time() * 2), 45 + Time.time());
                Draw.reset();
            }
```

Comment (ThePythonGuy3) [reply core/src/mindustry/world/blocks/defense/turrets/Turret.java:288]: I actually did it to avoid millisecond lag

**Comment by YellOw139** [comment]
I don't see why it's needed, so a toggle would be super nice, cuz personally I wouldn't like to see it.

BTW the target indicator shown in your screenshot is super ugly. Idea: Try to make it look like the mobile target indicator. It looks better. Simple and stylish.

**Comment by MEEPofFaith** [comment]
I might make a pr for that target stuff later...

**Comment by MEEPofFaith** [comment]
Created a PR for that.

**Comment by ThePythonGuy3** [comment]
Merged

**Comment by sk7725** [comment]
@ThePythonGuy3 can you use the mobile's player autotarget indicator? It would be more graaphically pleasing and be more consistent. If it looks too same to distinguish with the player's, you can maybe draw it in a different color or make it fade in and out(chuck a sin() in the alpha)

**Comment by MEEPofFaith** [comment]
> @ThePythonGuy3 can you use the mobile's player autotarget indicator? It would be more graaphically pleasing and be more consistent. If it looks too same to distinguish with the player's, you can maybe draw it in a different color or make it fade in and out(chuck a sin() in the alpha)

@sk7725 that's exactly what I did in the PR I did to his PR. I put the mobile autotarget into `Drawf.java`, then used `Drawf.target` for both the mobile autotarget and ThePythonGuy's turret targetting.

**Comment by sk7725** [comment]
cool

**Comment by ThePythonGuy3** [comment]
h

**Comment by Anuken** [comment]
After a discussion in Discord, I came to the conclusion that this isn't useful enough.

**Pull request closed without merging**